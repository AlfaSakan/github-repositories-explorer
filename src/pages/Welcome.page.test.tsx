import { describe, expect, it, vi } from "vitest";
import * as listUsersService from "../services/retrieveListUsers.service";
import * as listReposService from "../services/retrieveListRepos.service";
import { fireEvent, render, screen } from "@testing-library/react";
import { Welcome } from "./Welcome.page";

vi.spyOn(listUsersService, "retrieveListUsers").mockResolvedValue({
  incomplete_results: true,
  items: [
    {
      id: 1,
      login: "user 1",
      repos_url: "http://localhost:8080/",
    },
  ],
  total_count: 5,
});

vi.spyOn(listReposService, "retrieveListRepos").mockResolvedValue([
  {
    description: "description",
    id: 12,
    name: "name",
    owner: { id: 22 },
    stargazers_count: 12,
  },
]);

describe("Welcome", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Welcome />);

    fireEvent.input(screen.getByRole("textbox"));
    fireEvent.click(screen.getByText("Search"));

    expect(container).toMatchSnapshot();
  });
});
