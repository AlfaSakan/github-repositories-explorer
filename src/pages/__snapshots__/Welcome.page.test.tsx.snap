// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Welcome > should match the snapshot 1`] = `
<div>
  <main
    class="min-h-screen p-5 flex flex-col"
  >
    <form>
      <input
        class="border border-neutral-400 px-3 bg-neutral-200 w-full h-10 mb-3"
        id="name"
        name="name"
        placeholder="Enter Username"
        type="text"
        value=""
      />
      <button
        class="h-10 w-full bg-blue-500 text-white disabled:bg-neutral-400"
        disabled=""
        type="submit"
      >
        Search
      </button>
    </form>
    <div />
    <section
      aria-atomic="false"
      aria-label="Notifications Alt+T"
      aria-live="polite"
      aria-relevant="additions text"
      class="Toastify"
    />
  </main>
</div>
`;
